### DISCUSSION TITLE
CodeQL Kotlin 1.8 Support

### DISCUSSION THREAD

**Discussion by g-marshall-img-arena** [community/community#44817] [Code Security] [answered]
### Select Topic Area

Question

### Body

When upgrading our projects to Kotlin v1.8 this week, we started getting codeQL errors. Autorun was unable to run our builds all of a sudden. So we configured codeQL to manually build using the gradle wrapper. However, we still run into this error during that build process:  

`Kotlin version 1.8.0 is too recent. CodeQL currently supports versions below 1.7.30`

As a result, using codeQL for security checks is a blocker to us keeping our language version up to date, which is obviously not why we onboarded it. 

Does anyone know of a work around? Or is new version support planned anytime soon? 

Thanks in advance.

**Comment by smowton** [comment, marked as answer]
Support has been implemented and is anticipated to roll out starting around January 25th and finishing by February 15th.

To make sure you get support as soon as it gets released, you can opt into the latest release by using the `tools: latest` parameter to the `github/codeql-action/init` action. For example:

```
    - name: Initialize CodeQL
      uses: github/codeql-action/init@v2
      with:
        tools: latest
```

Note this slows the action down due to downloading the CodeQL bundle, so be sure to only use this if necessary, and schedule switching it off when no longer required (i.e., after February 15th).
